api_tests/scripts/authenticator.py

```diff
@@ -64,4 +65,14 @@ def get_code_from_provider(self, sign_in_response: 'response type') -> str:
         assert callback_response.status_code == 302, f"Callback request failed with {callback_response.status_code}"
 
         # Return code param from location header
-        return self.session.get_param_from_url(callback_response.headers.get('Location'), 'code')
\ No newline at end of file
+        return self.get_param_from_url(callback_response.headers.get('Location'), 'code')
+
+    @staticmethod
+    def get_params_from_url(url: str) -> dict:
+        """Returns all the params and param values from a given url as a dictionary"""
+        return dict(parse.parse_qsl(parse.urlsplit(url).query))
+
+    def get_param_from_url(self, url: str, param: str) -> str:
```

Comment:
Instead of this method we can use `get_params_from_url(my_url)[my_param]` which is more readable IMO than calling anther method just to reference the key in an object :)